documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +266,59 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 **Examples**: N/A
 
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S4002" />S4002	LocationMustHaveXmsMutability
+**Output Message**: Property 'location' must have '\"x-ms-mutability\":[\"read\", \"create\"]' extension defined. Resource Model: '{0}'
+
+**Description**: A tracked resource's `location` property must have the `x-ms-mutability` properties set as `read`, `create`.
+
+**Why the rule is important**: Location is a property that is set once and non-updatable for a tracked resource. Hence, per ARM guidelines the only operations allowed are `read` and `create`.
+
+**How to fix the violation**: Ensure that the `location` property in the tracked resource's hierarchy is correctly set to `read` and `create`.
```

Comment:
can you include an example of how the extension looks like or point to the extension's documentation? so someone could copy/paste?